src/backend/pipeline/modules/mosaicing.py

```diff
@@ -49,24 +50,38 @@ def run(self, data: Data) -> Data:
             # Make a mat
             stitched = Mat(stitched, data.input[0].channels) 
             data.modules[self.type]["stitched"] = stitched
+            data.persistable[self.type]["stitched"] = stitched.get()
 
             # split the image into equal patches for the segmentation module
-            # height and width of the patches
-            height = width = 512
-            channels = data.input[0].channels
-
-            # Calculate the number of patches to create
-            num_patches_horizontal = stitched.get().shape[1] // width
-            num_patches_vertical = stitched.get().shape[0] // height
-
-            patches: list[Mat] = []
-            # Loop through the image and extract each patch
-            for i in range(num_patches_vertical):
-                for j in range(num_patches_horizontal):
-                    patch = stitched.get()[i*height:(i+1)*height, j*width:(j+1)*width, :]
-                    patches.append(Mat(patch, channels))
-            # save the calculated patches for further usage
+            patches = self.create_patches(stitched, data.input[0].channels) 
             data.modules[self.type]["patches"] = patches
-            
+
+        # Upload persistable data to Google Cloud Storage
+        self.upload("terrafarm-example", data)
         # Run the next module
         return super().run(data)
+    
+    def create_patches(self, stitched: Mat, channels) -> list[Mat]:
```

Comment:
Let's add this to the documentation of the `Mosaicing` class and `Modules.MOSAICING`.